TramsDataApi/Gateways/NoticeToImproveGateway.cs

```diff
@@ -0,0 +1,103 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+using TramsDataApi.DatabaseModels;
+
+namespace TramsDataApi.Gateways
+{
+	public class NoticeToImproveGateway : INoticeToImproveGateway
+    {
+        private readonly TramsDbContext _tramsDbContext;
+        private readonly ILogger<NoticeToImproveGateway> _logger;
+
+        public NoticeToImproveGateway(TramsDbContext tramsDbContext, ILogger<NoticeToImproveGateway> logger)
+		{
+            _tramsDbContext = tramsDbContext;
+            _logger = logger;
+        }
+
+        public async Task<NoticeToImprove> CreateNoticeToImprove(NoticeToImprove request)
+        {
+            try
+            {
+                _tramsDbContext.NoticesToImprove.Add(request);
+                await _tramsDbContext.SaveChangesAsync();
+                return request;
+            }
+            catch (DbUpdateException ex)
+            {
+                _logger.LogError("Failed to create NoticeToImprove with Id {Id}, {ex}", request.Id, ex);
```

Comment:
The NoticeToImprove object wouldn't have an id unless it was successfully created, no? 
I appreciate this is probably done elsewhere.